prime-router/src/main/kotlin/azure/WorkflowEngine.kt

```diff
@@ -498,36 +492,44 @@ class WorkflowEngine(
                     .split()
                     .forEach {
                         val event = ReportEvent(Event.EventAction.SEND, it.id, actionHistory.generatingEmptyReport)
-                        this.dispatchReport(event, it, actionHistory, receiver, txn, context)
+                        this.dispatchReport(event, it, actionHistory, receiver, txn)
                     }
                 loggerMsg = "Queued to send immediately: HL7 split into ${report.itemCount} individual reports"
             }
             else -> {
                 val event = ReportEvent(Event.EventAction.SEND, report.id, actionHistory.generatingEmptyReport)
-                this.dispatchReport(event, report, actionHistory, receiver, txn, context)
+                this.dispatchReport(event, report, actionHistory, receiver, txn)
                 loggerMsg = "Queued to send immediately: ${event.toQueueMessage()}"
             }
         }
-        context?.logger?.info(loggerMsg)
+        logger.info(loggerMsg)
     }
 
     /**
      * The process step has failed. Ensure the actionHistory gets a 'warning' if it is not yet the 5th attempt
-     *  at this record. If it is the 5th attempt, set it to process_error
+     * at this record. If it is the 5th attempt, set it to process_error
+     * Also, save all the action history to the database.
      */
     fun handleProcessFailure(
         numAttempts: Int,
-        actionHistory: ActionHistory
+        actionHistory: ActionHistory,
+        message: String
     ) {
         // if there are already four process_warning records in the database for this reportId, this is the last try
-        val actionStatus = if (numAttempts >= 5) TaskAction.process_error else TaskAction.process_warning
-        // if count is < 5, add a process_warning status to the task
-        // if count is >= 5, add a process_error status to the task
-        actionHistory.setActionType(actionStatus)
-        actionHistory.trackActionResult(
-            "Failed to process $numAttempts times, setting status to $actionStatus."
-        )
-
+        if (numAttempts >= 5) {
+            actionHistory.setActionType(TaskAction.process_error)
+            actionHistory.trackActionResult(
+                "Failed to process $message $numAttempts times, setting status to process_error." +
+                    " Mo intervention required."
+            )
+            logger.fatal("$ALERT_TRIGGER: ${actionHistory.action.actionResult}")
+        } else {
```

Comment:
This is in a function only called by ProcessFunction, so this is a trigger of the super secret magic word, for ProcessFunction fatal failures.

I added it here because while ProcessFunction rarely fails, like SendFunction, it requires manual attention in the case of total failure.